hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/index/FlinkHoodieIndex.java

```diff
@@ -48,21 +47,22 @@ protected FlinkHoodieIndex(HoodieWriteConfig config) {
   @PublicAPIMethod(maturity = ApiMaturityLevel.DEPRECATED)
   public abstract List<WriteStatus> updateLocation(List<WriteStatus> writeStatuses,
                                                    HoodieEngineContext context,
-                                                   HoodieTable<T, List<HoodieRecord<T>>, List<HoodieKey>, List<WriteStatus>> hoodieTable) throws HoodieIndexException;
+                                                   HoodieTable hoodieTable) throws HoodieIndexException;
```

Comment:
Similar for engine-specific HoodieIndex classes to remove deprecated API methods altogether (in a separate PR).